earthpy/clip.py

```diff
@@ -83,34 +84,45 @@ def clip_line_poly(shp, clip_obj):
 
 
 def clip_shp(shp, clip_obj):
-   """A function to clip points, lines, polygon geometries based on an input
-   geometry.
+    """A function to clip points, lines, polygon geometries based on an input
+    geometry.
 
-   Both layers must be in the same Coordinate Reference System (CRS).
+    Both layers must be in the same Coordinate Reference System (CRS).
 
-   Depending on the geometry type, input data will be clipped to the full
-   extent of clip_obj using either clip_points or clip_line_poly.
+    Depending on the geometry type, input data will be clipped to the full
+    extent of clip_obj using either clip_points or clip_line_poly.
 
-   If there are multiple polygons in clip_obj,
-   data from shp will be clipped to the total boundary of
-   all polygons in clip_obj.
+    If there are multiple polygons in clip_obj,
+    data from shp will be clipped to the total boundary of
+    all polygons in clip_obj.
 
-   Parameters
-   ----------
-   shp : Geopandas dataframe
-         Vector layer (point, line, polygon) to be clipped to clip_obj.
+    Parameters
+    ----------
+    shp : Geopandas dataframe
+          Vector layer (point, line, polygon) to be clipped to clip_obj.
 
-   clip_obj : Geopandas dataframe
-         Polygon vector layer used to clip shp.
+    clip_obj : Geopandas dataframe
+          Polygon vector layer used to clip shp.
 
-   Returns
-   -------
-   Geopandas dataframe:
-        Vector data (points, lines, polygons) from shp clipped to
-        polygon boundary from clip_obj.
+    Returns
+    -------
+    Geopandas dataframe:
+         Vector data (points, lines, polygons) from shp clipped to
+         polygon boundary from clip_obj.
     """
 
+    # Both objects should be GeoDataFrame
+    try:
+        shp.geometry
+        clip_obj.geometry
+    except (AttributeError, TypeError):
+        raise AssertionError('Input variables should be GeoDataFrames with a geometry column')
+
+    # Test to ensure shapes intersect
+    if not any(shp.intersects(clip_obj)):
+        raise ValueError("Shape and crop extent do not overlap.")
+
     if shp["geometry"].iloc[0].type == "Point":
```

Comment:
@lwasser is it possible in geopandas to have a data frame where some rows (e.g., the first) contain points, and other rows contain lines or polygons?

Reply:
@mbjoseph i'm fairly certain the answer for a GeoDataFrame is no given it's based on the shapefile structure. The idea is that a single object contains one data type - points OR lines or Polys. Thus my guess is if you try to feed a GDF both it will be super confused. NOW with that said i suppose it is possible to have a line that is actually a POLY but created as a line object. Think county boundaries for instance. sometimes that will be a line feature even though it should be a poly.  Are you suggesting another test for something like that? Or just ensuring == "Point" is always true! ?

Reply:
Ah cool @lwasser! I wasn't sure whether just checking the first element of the 'geometry' series was going to be robust enough. But, it sounds like if the first element is a "Point", then the rest must be.

Reply:
Yes -- i would assume that would be the case all of the time given the rules of "shapefile" and GDF structure. If it was a geodatabase i would not assume that. it's a great thing to check tho - i appreciate the comment!